src/GraphQL/ObjectExtensions.cs

```diff
@@ -123,7 +124,15 @@ public static Type GetInterface(this Type type, string name)
         public static object ConvertValue(object value, Type fieldType)
         {
             if (value == null) return null;
-            if (fieldType == typeof(DateTime) && value is DateTime) return value;
+
+            if (fieldType == typeof(DateTime))
```

Comment:
I did the change and it works. I agree that logic  should be in the DateTimeConverter class. Thanks!